project-base/docker/nginx/nginx-google-cloud.conf

```diff
@@ -0,0 +1,66 @@
+upstream php-upstream {
+    server php-fpm:9000;
+}
+
+server {
+    listen 8080;
+    access_log /var/log/nginx/shopsys-framework.access.log;
+    root /var/www/html/web;
+    server_tokens off;
+
+    location ~ /\. {
+        # hide dotfiles (send to @app)
+        try_files @app @app;
+    }
+    location ~ ^/index.php$ {
+        # hide index.php (send to @app)
+        try_files @app @app;
+    }
+    location / {
+        # try to serve existing files directly, fallback to @app
+        try_files $uri @app;
+
+        # Enable compression of JavaScripts and CSS
+        location ~ (\.js|\.css)$ {
+            gzip on;
+            gzip_types application/javascript text/css;
+
+            # Cache versioned static content for 1 year
+            location ~ ((-v[0-9]+\.js)|(_[0-9]+\.css))$ {
+                expires 1y;
+
+                # (try_files directive is not inherited from the parent location so we have to repeat it)
+                try_files $uri @app;
+            }
+
+            # (try_files directive is not inherited from the parent location so we have to repeat it)
+            try_files $uri @app;
+        }
+
+        location ~ ^/content(-test)?/images/ {
+            # Newly uploaded images get new ID (different URL) so they could be cached forever.
+            # But change of resolution in images.yml does not induce a change of URL
+            # so it is safer to cache generated images only for few days.
+            expires 1w;
+
+            # (try_files directive is not inherited from the parent location so we have to repeat it)
+            try_files $uri @app;
+        }
+
+
+        location ~ ^/content/ {
+            resolver 8.8.8.8;
+            proxy_pass https://storage.googleapis.com/{{GOOGLE_CLOUD_STORAGE_BUCKET_NAME}}/web$request_uri;
```

Comment:
is there something to do to have google cloud bucket publicly accessible? if yes please write something about it in docs.

Reply:
https://erikzaadi.com/2014/06/23/handle-proxy-404-in-nginx/